data/account/participation.go

```diff
@@ -61,31 +61,36 @@ type Participation struct {
 type participationIDData struct {
 	_struct struct{} `codec:",omitempty,omitemptyarray"`
 
-	Parent      basics.Address    `codec:"addr"`
-	VRFSK       crypto.VrfPrivkey `codec:"vrfsk"`
-	FirstValid  basics.Round      `codec:"fv"`
-	LastValid   basics.Round      `codec:"lv"`
-	KeyDilution uint64            `codec:"kd"`
+	transactions.KeyregTxnFields
```

Comment:
I think that the previous implementation was fine - the identity of the participation key is not really related to the key registration transaction fields. ( it happened to have a similar set of fields, but it's a different "creature" ).

Reply:
My thinking was that Idan's project is likely to change both of these, and maybe this would allow the change to happen in one place.

Interestingly, the existing ID doesn't include key dilution. I'm not sure if that's a bug, but including it here definitely gives more entropy to the ID.

Reply:
This ParticipationID hash need to capture the "identity" of the file.. but that's about it. I don't think we need to worry too much about Idan's upcoming changes, as there would probably mean adding one or two more fields to the `transactions.KeyregTxnFields` which we *can* add to the ParticipationID, but don't have to.